auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -262,6 +267,93 @@ public static int getDefaultTheme() {
         return R.style.FirebaseUI;
     }
 
+    /**
+     * Signs the user in without any UI if possible. If this operation fails, you can safely start a
+     * UI-based sign-in flow knowing it is required.
+     *
+     * @param context        requesting the user be signed in
+     * @param desiredConfigs to use for silent sign in. Only Google and email are currently
+     *                       supported, the rest will be ignored.
+     * @return a task which indicates whether or not the user was successfully signed in.
+     */
+    @NonNull
+    public Task<AuthResult> silentSignIn(@NonNull Context context,
+                                         @NonNull List<IdpConfig> desiredConfigs) {
+        if (mAuth.getCurrentUser() != null) {
+            throw new IllegalArgumentException("User already signed in!");
+        }
+
+        List<IdpConfig> configs = new ArrayList<>();
+        for (IdpConfig config : desiredConfigs) {
+            String provider = config.getProviderId();
+            if (provider.equals(EmailAuthProvider.PROVIDER_ID)
+                    || provider.equals(GoogleAuthProvider.PROVIDER_ID)) {
+                configs.add(config);
+            }
+        }
+
+        if (configs.isEmpty()) {
+            throw new IllegalArgumentException("No supported providers were supplied. " +
+                    "Add either Google or email support.");
+        }
+
+        final Context appContext = context.getApplicationContext();
+        final IdpConfig google =
```

Comment:
This is kind of a funny way to get here:

  * Run through desired configs, filter out everything besides `google` and `email`
  * Then use `getConfigFromIdps` to get `google` and `email` out of the filtered list

Why not just use `getConfigfromIdps` on `desiredConfigs` and throw the `IllegalArgumentException` if `google` and `email` both end up null?

Reply:
Lol, yeah, that's a much better way to do it.